src/arrayMethodSort.js

```diff
@@ -4,8 +4,24 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
+  [].__proto__.sort2 = function (compareFunction) {
     // write code here
+    for (let i = 0; i < this.length; i++) {
+      for (let j = i + 1; j < this.length; j++) {
+        const compareResult = compareFunction
+          ? compareFunction(this[i], this[j]) > 0
+          : String(this[i]) > String(this[j]);
+
+        if (compareResult > 0 || compareResult) {
+          const temp = this[i];
+
+          this[i] = this[j];
+          this[j] = temp;
+        }
+      }
+    }
```

Comment:
You are using a bubble sort algorithm, which is known for its simplicity but not for its efficiency. It has a time complexity of O(n^2), which is not ideal for large arrays. If performance is a concern, consider implementing a more efficient sorting algorithm like quicksort or mergesort.